networkx/readwrite/json_graph/tests/test_node_link.py

```diff
@@ -44,6 +47,21 @@ def test_multigraph(self):
         nx.is_isomorphic(G, H)
         assert_equal(H[1][2]['second']['color'], 'blue')
 
+    def test_graph_with_tuple_nodes(self):
+        G = nx.Graph()
+        G.add_edge((0, 0), (1, 0), color=[255, 255, 0])
+        d = node_link_data(G)
+        with NamedTemporaryFile('w', delete=False) as file:
```

Comment:
Let's just use json.dumps and json.loads here to avoid writing a temporary file.